app/lib/katello/util/errata.rb

```diff
@@ -46,13 +46,18 @@ def filter_by_state(errata_list, errata_state)
         end
       end
 
-      def filter_errata_by_pulp_href(errata, package_pulp_hrefs)
+      def filter_errata_by_pulp_href(errata, package_pulp_hrefs, source_repo_rpm_filenames)
         return [] if package_pulp_hrefs.empty?
         rpms = Katello::Rpm.where(:pulp_id => package_pulp_hrefs)
         rpm_filenames = rpms.map { |rpm| File.basename(rpm.filename) }
+        source_repo_rpm_filenames = source_repo_rpm_filenames.map { |rpm| File.basename(rpm) }
         matching_errata = []
         errata.each do |erratum|
-          if erratum.packages.any? && (erratum.packages.pluck(:filename) - rpm_filenames).empty?
+          # The erratum should be copied if package_pulp_hrefs has all of its packages that are available in the source repo.
+          next if erratum.packages.empty?
+          rpms_in_erratum_and_source_repo = erratum.packages.pluck(:filename) & source_repo_rpm_filenames
+          next if rpms_in_erratum_and_source_repo.empty?
```

Comment:
do these to 'next' statement ignore empty errata?  I think we have to support those

Reply:
ignore this comment!